lib/bolt/transport/winrm.rb

```diff
@@ -128,9 +128,12 @@ def run_task(target, task, arguments, _options = {})
                 # fortunately, using PS with stdin input_method should never happen
                 if input_method == 'powershell'
                   conn.execute(<<-PS)
-$private:taskArgs = Get-ContentAsJson (
+$private:tempArgs = Get-ContentAsJson (
   $utf8.GetString([System.Convert]::FromBase64String('#{Base64.encode64(JSON.dump(arguments))}'))
 )
+$allowedArgs = (Get-Command "#{remote_path}").Parameters.Keys
+$private:taskArgs = @{}
+$private:tempArgs.Keys | ? { $_ -in $allowedArgs } | % { $private:taskArgs[$_] = $private:tempArgs[$_] }
```

Comment:
Just double checked the `-in` operator. It's PS3 only, so we might want to change `$_ -in $allowedArgs` to `$allowedArgs -contains $_`

Reply:
Makes sense.